table/sst_file_writer.cc

```diff
@@ -78,11 +82,20 @@ struct SstFileWriter::Rep {
       return builder->status();
     }
 
+    std::string user_key_buf;
+    Slice effective_user_key = user_key;
+    if (strip_timestamp) {
+      // Logically remove the timestamp while the table builder will later
+      // remove it physically when data block is being built.
+      AppendUserKeyWithMinTimestamp(&user_key_buf, user_key, ts_sz);
+      effective_user_key = user_key_buf;
+    }
```

Comment:
I'm wondering if it would be more intuitive to expect clients to provide keys with the min timestamp to start with (instead of us overwriting the timestamp)?

Reply:
That's a great point! Although users are expected to provide timestamps, the APIs usage will be more intuitive if they are only allowed to provide a min timestamp when the `persist_user_defined_timestamps` flag is false.